plugins/local_file/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+add_custom_target(all-plugins-local_file)
+add_dependencies(all-plugins all-plugins-local_file)
+
+set(
+  IRODS_EXPERIMENTAL_API_PLUGINS
+  main
+)
+
+foreach(plugin IN LISTS IRODS_EXPERIMENTAL_API_PLUGINS)
+  string(TOUPPER ${plugin} PLUGIN_UPPERCASE)
+  set(plugin_target "irods_http_plugin-${plugin}")
+  set(plugin_name "${plugin}")
+
+  add_library(
+    ${plugin_target}
```

Comment:
Does this produce a .so called `libirods_http_plugin-main.so`?

If so, we'll need to come up with a new name (i.e. replace `main`).

Reply:
Currently, it produces `libmain.so`: https://github.com/irods/irods_client_http_api/pull/331#pullrequestreview-2266973777

If I understand correctly, it gets its name based off the source file.

Reply:
Try to name `-local_file` and `-provider_claim`.

Reply:
Plugin names updated.